sklearn/neural_network/tests/test_stochastic_optimizers.py

```diff
@@ -21,10 +21,11 @@ def test_base_optimizer():
 
 def test_sgd_optimizer_no_momentum():
     params = [np.zeros(shape) for shape in shapes]
+    rng = np.random.RandomState(0)
 
     for lr in [10 ** i for i in range(-3, 4)]:
         optimizer = SGDOptimizer(params, lr, momentum=0, nesterov=False)
-        grads = [np.random.random(shape) for shape in shapes]
+        grads = [rng.random(shape) for shape in shapes]
```

Comment:
`RandomState` has a slightly different api:

```suggestion
        grads = [rng.random_sample(shape) for shape in shapes]
```

(Side note: the new numpy [Generator](https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.random.html#numpy.random.Generator.random) supports `random`)

Reply:
Hum, good catch. I was not aware of this discrepancy.

Weird I thought I ran the tests locally before pushing...

Reply:
It probably depends on the version of numpy.